python/ml4ir/base/pipeline.py

```diff
@@ -439,6 +436,13 @@ def run(self):
 
         return experiment_tracking_dict
 
+    def pre_processing_step(self):
+        """
+        Performs any pre processing step (such as copying data) by custom class inheriting this class, which would
```

Comment:
this is a somewhat cryptic docstring. I would go with something like: "Performs arbitrary pre-processing steps such as copying or transforming data that the rest of the code can not accommodate. It serves as a placeholder without an explicit implementation (returns `self`) in the `base` pipeline. We expect that users can extend it in their custom pipelines."